limacharlie/retention.go

```diff
@@ -89,3 +89,102 @@ func (org *Organization) EventByAtom(sensorID, atom string) (EventContainer, err
 	err := org.client.reliableRequest(http.MethodGet, fmt.Sprintf("insight/%s/%s/%s", org.client.options.OID, sensorID, atom), q)
 	return event, err
 }
+
+type HistoricalDetectionsRequest struct {
+	// Cat is the category of the detections to fetch
+	Cat string `json:"cat"`
+	// Cursor is optional for paginated access, set to '-' for first query
+	Cursor string `json:"cursor"`
+	// Start is the required timestamp in seconds where to stop fetching detections
+	Start int `json:"start"`
+	// End is the required timestamp in seconds where to stop fetching detections
+	End int `json:"end"`
+	// Limit maximum number of detections to return
+	Limit int `json:"limit"`
+}
+
+func (org Organization) HistoricalDetections(detectionReq HistoricalDetectionsRequest) (HistoricalDetectionsResponse, error) {
+
+	var results HistoricalDetectionsResponse
+
+	if detectionReq.Cursor == "" {
+		detectionReq.Cursor = "-"
+	}
+
+	q := makeDefaultRequest(&results)
+	q = q.withQueryData(Dict{
+		"cat":    detectionReq.Cat,
+		"cursor": detectionReq.Cursor,
+		"start":  detectionReq.Start,
+		"end":    detectionReq.End,
+		"limit":  detectionReq.Limit,
+	})
```

Comment:
You don't need to convert it to a dict, the `withQueryData()` supports the JSON Marshalling interface.
```suggestion
	q = q.withQueryData(detectionReq)
```